runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -23,13 +23,94 @@
+
+   for (i = TR::RealRegister::x8; i <= TR::RealRegister::x15; i++)
+      _properties._registerFlags[i] = 0; // x8 - x15 volatile
+
+   _properties._registerFlags[TR::RealRegister::x16]   = ARM64_Reserved; // IP0
+   _properties._registerFlags[TR::RealRegister::x17]   = ARM64_Reserved; // IP1
+
+   for (i = TR::RealRegister::x18; i <= TR::RealRegister::x28; i++)
+      _properties._registerFlags[i] = Preserved; // x18 - x28 Preserved
+
+   _properties._registerFlags[TR::RealRegister::x29]   = ARM64_Reserved; // FP
+   _properties._registerFlags[TR::RealRegister::x30]   = ARM64_Reserved; // LR
+   _properties._registerFlags[TR::RealRegister::sp]    = ARM64_Reserved;
+   _properties._registerFlags[TR::RealRegister::xzr]   = ARM64_Reserved;
+
+   _properties._registerFlags[TR::RealRegister::v0]    = FloatArgument|FloatReturn;
+   _properties._registerFlags[TR::RealRegister::v1]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v2]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v3]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v4]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v5]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v6]    = FloatArgument;
+   _properties._registerFlags[TR::RealRegister::v7]    = FloatArgument;
+
+   for (i = TR::RealRegister::v8; i <= TR::RealRegister::LastFPR; i++)
+      _properties._registerFlags[i] = 0; // v8 - v31 volatile
+
+   _properties._numIntegerArgumentRegisters  = 8;
+   _properties._firstIntegerArgumentRegister = 0;
+   _properties._numFloatArgumentRegisters    = 8;
+   _properties._firstFloatArgumentRegister   = 8;
+
+   _properties._argumentRegisters[0]  = TR::RealRegister::x0;
+   _properties._argumentRegisters[1]  = TR::RealRegister::x1;
+   _properties._argumentRegisters[2]  = TR::RealRegister::x2;
+   _properties._argumentRegisters[3]  = TR::RealRegister::x3;
+   _properties._argumentRegisters[4]  = TR::RealRegister::x4;
+   _properties._argumentRegisters[5]  = TR::RealRegister::x5;
+   _properties._argumentRegisters[6]  = TR::RealRegister::x6;
+   _properties._argumentRegisters[7]  = TR::RealRegister::x7;
+   _properties._argumentRegisters[8]  = TR::RealRegister::v0;
+   _properties._argumentRegisters[9]  = TR::RealRegister::v1;
+   _properties._argumentRegisters[10] = TR::RealRegister::v2;
+   _properties._argumentRegisters[11] = TR::RealRegister::v3;
+   _properties._argumentRegisters[12] = TR::RealRegister::v4;
+   _properties._argumentRegisters[13] = TR::RealRegister::v5;
+   _properties._argumentRegisters[14] = TR::RealRegister::v6;
+   _properties._argumentRegisters[15] = TR::RealRegister::v7;
+
+   _properties._firstIntegerReturnRegister = 0;
+   _properties._firstFloatReturnRegister   = 1;
+
+   _properties._returnRegisters[0]  = TR::RealRegister::x0;
+   _properties._returnRegisters[1]  = TR::RealRegister::v0;
+
+   _properties._numAllocatableIntegerRegisters = 27;
```

Comment:
Should be `25` if you exclude the vmthread and the Java stack pointer.

Reply:
Changed the number.